core/data/src/main/java/uk/gov/gchq/gaffer/data/elementdefinition/view/NamedView.java

```diff
@@ -0,0 +1,196 @@
+    }
+
+    public void setName(final String viewName) {
+        this.name = viewName;
+    }
+
+    public String getName() {
+        return this.name;
+    }
+
+    public void setMergedNamedViewNames(final List<String> mergedNamedViewNames) {
+        if (mergedNamedViewNames != null) {
+            if (this.mergedNamedViewNames != null) {
+                this.mergedNamedViewNames.addAll(mergedNamedViewNames);
+            } else {
+                this.mergedNamedViewNames = mergedNamedViewNames;
+            }
+        }
+    }
+
+    public List<String> getMergedNamedViewNames() {
+        return mergedNamedViewNames;
+    }
+
+    public void setParameters(final Map<String, Object> parameters) {
+        if (parameters != null) {
+            if (null != this.parameters) {
+                this.parameters.putAll(parameters);
+            } else {
+                this.parameters = parameters;
+            }
+        }
+    }
+
+    public Map<String, Object> getParameters() {
+        return parameters;
+    }
+
+    @Override
+    public boolean canMerge(final View addingView, final View srcView) {
+        if (addingView instanceof NamedView && !(srcView instanceof NamedView)) {
+            if (((NamedView) addingView).getName() != null) {
+                if (!((NamedView) addingView).getName().isEmpty()) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    public abstract static class BaseBuilder<CHILD_CLASS extends BaseBuilder<?>> extends View.BaseBuilder<CHILD_CLASS> {
+
+        public BaseBuilder() {
+            this(new NamedView());
+        }
+
+        public BaseBuilder(final NamedView namedView) {
+            super(namedView);
+        }
+
+        public CHILD_CLASS name(final String name) {
+            getElementDefs().setName(name);
+            return self();
+        }
+
+        public CHILD_CLASS parameters(final Map<String, Object> parameters) {
+            getElementDefs().setParameters(parameters);
+            return self();
+        }
+
+        @Override
+        @JsonIgnore
+        public CHILD_CLASS json(final InputStream... inputStreams) throws SchemaException {
+            return json(NamedView.class, inputStreams);
+        }
+
+        @Override
+        @JsonIgnore
+        public CHILD_CLASS json(final Path... filePaths) throws SchemaException {
+            return json(NamedView.class, filePaths);
+        }
+
+        @Override
+        @JsonIgnore
+        public CHILD_CLASS json(final byte[]... jsonBytes) throws SchemaException {
+            return json(NamedView.class, jsonBytes);
+        }
+
+        @Override
+        public CHILD_CLASS merge(final View view) {
+            if (null != view) {
+                if (view instanceof NamedView) {
+                    NamedView namedViewInstance = (NamedView) view;
+                    if (null != namedViewInstance.getName() && !namedViewInstance.getName().isEmpty()) {
```

Comment:
This is a bit tricky to read - perhaps use StringUtils.isEmpty(namedViewInstance.getName()) to simplify your logic.